src/runtime/virtcontainers/qemu.go

```diff
@@ -1985,16 +1965,13 @@ func (q *qemu) hotplugMemory(memDev *MemoryDevice, op Operation) (int, error) {
 		return 0, nil
 	case AddDevice:
 		memLog.WithField("operation", "add").Debugf("Requested to add memory: %d MB", memDev.SizeMB)
-		maxMem, err := q.hostMemMB()
-		if err != nil {
-			return 0, err
-		}
 
 		// Don't exceed the maximum amount of memory
-		if currentMemory+memDev.SizeMB > int(maxMem) {
-			// Fixme: return a typed error
-			return 0, fmt.Errorf("Unable to hotplug %d MiB memory, the SB has %d MiB and the maximum amount is %d MiB",
-				memDev.SizeMB, currentMemory, maxMem)
+		if currentMemory+memDev.SizeMB > int(q.config.DefaultMaxMemorySize) {
+			memDev.SizeMB = int(q.config.DefaultMaxMemorySize) - currentMemory
+			if memDev.SizeMB == 0 {
```

Comment:
Is `memDev.SizeMB < 0` considerable?

Reply:
Theoretically we'd never ever hotplug more than DefaultMaxMemorySize, thus  current Memory would, in the worst case scenario, be equal to DefaultMaxMemorySize.

But adding a `<=` check doesn't hurt, so I'm doing it.  Thanks for the suggestion!